src/main/resources/site/filters/favicon.js

```diff
@@ -4,10 +4,17 @@ var thymeleaf = require('/lib/xp/thymeleaf');
 exports.responseFilter = function(req, res) {
     var siteConfig = portal.getSiteConfig();
     var imageId = siteConfig.favicon;
-    if(!imageId) {
+    if(
+      !imageId ||
+      (res.headers && res.headers.serverSideDisableFavicon)
+    ) {
         return res;
     }
 
+    if(res.headers && res.headers.serverSideDisableFavicon) {
+        delete res.headers.serverSideDisableFavicon; // Strip serverSide data so the end user never receives them.
+    }
```

Comment:
The only way this code will ever run is if the `imageId` is not set (if you haven't uploaded an image yet).

If an image IS uploaded, the `serverSideDisableFavicon` header will be sent to the client after all (since it will return on line 8/11). You would either need to delete the header there as well, or do something like this:

```
var disableFavicon = res.headers && res.headers.serverSideDisableFavicon;
delete res.headers.serverSideDisableFavicon;
if (!imageId || disableFavicon) {
  return res;
}

var view = ..... // rest of code.
```